README.md

```diff
@@ -1,78 +1,89 @@
+     https://github.com/grpc/proposal/pulls?q=is%3Apr+sort%3Acreated-desc)).
+     Find the first _new_ proposal PR of the same type, and use the following
+     number.
+1. Submit a Pull Request.  The PR description should be formatted as follows:
+
+        $CategoryName##: <title>
+
+1. Someone from the gRPC team will be assigned as an APPROVER as part of this
+review.
+1. For a period of at least 10 business days (the minimum comment period), it is
+expected that the OWNER will respond to the comments and make updates to the RFC
+as new commits to the PR. The OWNER is encouraged to solicit as much feedback on
+the proposal as possible during this period.
 1. If there is consensus as deemed by the APPROVER during the comment period,
-the APPROVER will mark the proposal as final and assign it a gRFC number.
-Once this is assigned (as part of the closure of discussion), the OWNER will
-update the state of the PR as final and submit the PR.
-Commits must not be squashed; the commit history serves as a log of changes
-made to the proposal.
+the APPROVER will approve the PR on GitHub. The PR will then be merged by either
+the OWNER, if possible, or the APPROVER otherwise.
+
+All proposals merged into the repo are considered "approved" and either
+implemented or ready to implement.
 
 ## APPROVER
+
 - By default ``a11r`` is the approver unless another approver is assigned
 on a per-proposal basis.
 - If the assigned APPROVER and the OWNER cannot satisfactorily settle an issue,
 the final APPROVER is still ``a11r``.
 
 ## Proposal Categories
+
 The proposals shall be numbered in increasing order.
 
-- ``#An`` - Affects all languages.
-- ``#Pnn`` - Affects processes, such as the proposal process itself.
-- ``#Lnnn`` - Language specific changes to external APIs or platform support.
-- ``#Gnnnn`` - Protocol level changes.
-
-## Proposal Status
-1. Every uncommitted proposal candidate starts off in the ``Draft`` state.
-1. After it accepted for review and posted to the group, it enters the
-``In Review`` state.
-1. Once it is approved for submission by the arbiter, it goes into the
-``Final`` state. Only minor changes are allowed (what qualifies as minor is
-left to the APPROVER).
-1. If a proposal needs to be revisited, it can be moved back to the ``Draft``
-or ``In Review`` state. This can happen if issues are discovered during
-implementation. At which point, the review process as described above must be
-followed.
-1. Once a proposal is ``Final`` and if it has been implemented by a language,
-it can be updated to a status of ``Implemented`` with the implementing
-languages listed. (Listing versions is not required.)
+- ``An`` - Affects all languages.
+- ``Pnn`` - Affects processes, such as the proposal process itself.
+- ``Lnnn`` - Language specific changes to external APIs or platform support.
+- ``Gnnnn`` - Protocol level changes.
+
+## Updating proposals
+
+Sometimes small changes are needed to approved proposals.  Rather than create
```

Comment:
I do think we should have some hints for what "small" is.

> Sometimes small changes are needed to approved proposals. For example, editorial fixes, clarifying text, and detailed fixes noticed during implementation or usage. Rather than create new proposals for such changes, it is often better to revise the existing one.

Reply:
Honestly I think anything that's not considered a "major" change should be a modification of a gRFC instead of a new one.  A major change would be like an entirely different API or a wholly different mechanism for solving the same problem.  "Minor" as described below, IMO, is for editorial fixes and clarifying text, which is allowed to be approved without the 10 day waiting period.

Reply:
This is the original `README.md` text, for reference:

1. Once it is approved for submission by the arbiter, it goes into the ``Final`` state. Only minor changes are allowed (what qualifies as minor is left to the APPROVER).
1. If a proposal needs to be revisited, it can be moved back to the ``Draft`` or ``In Review`` state. This can happen if issues are discovered during implementation. At which point, the review process as described above must be followed.

Reply:
Oh, so that's where you got that language... I think then we should harken back to the original language.

> Only minor updates typically occur after a proposal is approved (and merged?). Minor updates do not require a comment period. What qualifies as minor is left to the APPROVER. If an update is needed and is not minor, then the process's 10 day comment period must be observed. This can happen if issues are discovered during implementation.
>
> When updating a proposal, the PR description should be named as follows:
>    $CategoryName## update: <description of change>

This is potentially more restrictive, but it also doesn't change what updates can occur and how compared to the existing text.

Reply:
> Only minor updates typically occur after a proposal is approved

Sorry, but I don't like your wording.  "Only minor updates typically occur" -- to say "only X is typical" is unusual and unnecessary.  The goal of the original process here was:

1. "Minor" changes to a gRFC are allowed with no 10-day discussion period.
2. Bigger changes to a gRFC are allowed with the 10-day discussion period.

I believe I expressed this adequately with my wording.  I've made a small change here to move stuff / simplify.  LMK if you disagree.

Reply:
I disagree with your interpretation of the original text. I think the "This can happen if issues are discovered during implementation" part is important, as it provides a scope/context of the type of change we're talking about. We're not talking about redesigning old gRFCs 5 years after it was implemented because we no longer like that approach. Such things should be a new gRFC. Your proposed text _encourages_ changing existing gRFCs, with no real limit on what sort of changes we'd expect to make.

Reply:
If a gRFC has been implemented we probably aren't going to be materially modifying it in place except possibly to correct a problem discovered later or add a necessary, small tweak.  I take the "This can happen if" as an example, not to mean "This can _only_ happen if".  IMO we should leave it up to our judgement on a case-by-case basis whether it's better to update a gRFC or create a new one.  That's realistically already what we are doing, and I don't see this minor wording change (done to remove now-unnecessary clauses with the removal of the "Status" field) affecting anything.

@markdroth, thoughts?

Reply:
> I take the "This can happen if" as an example, not to mean "This can only happen if".

I agree. But the example does provide a _guide_. If someone wanted to do a huge change we could say, "that's a bigger change than we'd expect for an approved gRFC" and they wouldn't be surprised, because you can see how far of a departure that huge change is from the example. That is gone in the new language, and the new language _encourages_ making the change to the existing gRFC without any limit, implied or explicit.

Reply:
I can see both sides of this.  I agree with Doug that we should ultimately leave the decision to the approver.  But I also agree with Eric that we don't want to leave this completely open-ended; I think we do want to make it clear that non-minor changes to an existing gRFC are discouraged.

I will note that we do have one example where we made significant changes to a gRFC after it was already implemented, which was the modifications to A51 in https://github.com/grpc/proposal/pull/345.  In hindsight, this really should have been done as a separate gRFC; I was the one who suggested modifying the existing gRFC, since I thought it would be a fairly simple change, but it turned out to be more complex than I expected.  I think that having written guidance discouraging that might have prompted someone to object before that PR got merged.

In general, my feeling is that any modification to an existing gRFC should be minor enough that it doesn't require the 10-day comment period.  I think that in most cases, if we're considering needing the 10-day comment period, that's probably a signal that we should be writing a new gRFC rather than modifying an existing one.  But I don't want to be prescriptive about this -- I do think we want to allow the approver to use their judgement.

I suggest the following wording:

"""
Sometimes changes are needed to already-merged proposals, such as when we discover problems during implementation.  Rather than create new proposals for such changes, it is often better to revise the existing one.  However, this approach may not be appropriate for larger changes, especially after the design has already been implemented in most languages.

To make changes to an already-merged proposal, a PR may be approved and merged at the OWNER's and APPROVER's convenience.  Note that the APPROVER will decide whether this kind of post-merge change is appropriate.  The approver will also determine whether the 10-day comment period is necessary, based on how significant the change is.
"""

Reply:
Seems fine to me.  Updated.